components/messaging/GroupChat.tsx

```
@@ -0,0 +1,1071 @@
+          // Promote member to moderator
+          await database.write(async () => {
+            const member = members.find(m => m.userId === memberId);
+            if (member) {
+              await member.promoteToModerator();
+            }
+          });
+
+          // Update local state - reload from database to get updated model
+          const promotedMembers = await database.get('group_members')
+            .query(Q.where('group_id', groupId))
+            .fetch();
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)